apps/rule-manager/client/src/ts/components/RuleForm.tsx

```diff
@@ -45,6 +45,7 @@ export const RuleForm = ({onRuleUpdate, ruleData, setRuleData, createRuleFormOpe
     }) => {
     const [showErrors, setShowErrors] = useState(false);
     const [errors, setErrors] = useState<FormError[]>([]);
+    const pageData = useContext(PageContext)
```

Comment:
Non-blocking: just thinking that we do this in two places – we could make this slightly more concise by writing a `useCreateEditPermissions` hook that looks a bit like:

```ts
const useCreateEditPermissions = () => {
  const pageData = useContext(PageContext);
  // Do not recalculate permissions if the permissions list has not changed
  return useMemo(() => hasCreateEditPermissions(pageData.permissions), [pageData.permissions]);
}
```

(The memoisation optimisation is not super important, but arguably a good habit when calling pure fns from React components.)

Reply:
Now done.